resolver/resolver_test.go

```diff
@@ -56,19 +70,31 @@ func TestRecurivePathResolution(t *testing.T) {
 		t.Fatal(err)
 	}
 
-	resolver := resolver.NewBasicResolver(dagService)
-	node, err := resolver.ResolvePath(ctx, p)
+	fetcherFactory := bsfetcher.NewFetcherConfig(bsrv)
+	fetcherFactory.NodeReifier = unixfsnode.Reify
+	fetcherFactory.PrototypeChooser = dagpb.AddSupportToChooser(func(lnk ipld.Link, lnkCtx ipld.LinkContext) (ipld.NodePrototype, error) {
+		if tlnkNd, ok := lnkCtx.LinkNode.(schema.TypedLinkNode); ok {
+			return tlnkNd.LinkTargetNodePrototype(), nil
+		}
+		return basicnode.Prototype.Any, nil
+	})
+	resolver := resolver.NewBasicResolver(fetcherFactory)
+
+	node, lnk, err := resolver.ResolvePath(ctx, p)
 	if err != nil {
 		t.Fatal(err)
 	}
 
+	uNode, ok := node.(unixfsnode.PathedPBNode)
+	require.True(t, ok)
+	fd := uNode.FieldData()
```

Comment:
UnixFSNode the repo should probably be renamed something along the lines of "ADLs for DagPB" because it's not entirely just UnixFS.

"PathedPBNode" is an ADL that's not quite UnixFS but also not just DagPB with no additional logic.

Basically, it turns a PB node into a map where the keys are the Link names. 

We end up needing this for path traversal even when we're not dealing with unixFS.

Perhaps this particular ADL should not live in the go-unixfsnode repo. I don't particularly feel like it's pressing to move it right this second though.

Reply:
IIUC this is interesting. It seems like we currently (i.e. before the IPLD prime PRs) support pathing through DagPB (not UnixFS) data by name even when ignoring the UnixFS resolver. This also seems to preclude any ability to address by path the `Data` fields in DagPB nodes or the `Size` fields in their links.

AFAICT this seems like a mistake since it prevents some of the DagPB data from being addressable by pathing. This is likely something we'll need to resolve as part of https://github.com/ipfs/go-ipfs/pull/7976#issuecomment-895997465.